Objective-C/CBLIndexConfiguration.h

```diff
@@ -0,0 +1,40 @@
+//
+//  CBLIndexConfiguration.h
+//  CouchbaseLite
+//
+//  Copyright (c) 2021 Couchbase, Inc All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface CBLIndexConfiguration : NSObject
+- (instancetype) init NS_UNAVAILABLE;
+
+@end
+
+@protocol CBLIndexConfigurationProtocol <NSObject>
+- (instancetype) initWithExpression: (NSString*)expression;
+
+@end
+
+@interface CBLFullTextIndexConfiguration: CBLIndexConfiguration <CBLIndexConfigurationProtocol>
```

Comment:
^^ ValueIndexConfiguration & CBLFullTextIndexConfiguration, kept it in same file. From import perspective, its better to keep it as separate file

Reply:
It's missing some properties.